roles/satellite/tasks/internals/product-definition.yml

```diff
@@ -33,9 +49,11 @@
     product: "{{ item.0.product }}"
     name: "{{ item.1.name }}"
     url: "{{ item.1.url | default('') }}"
+    gpg_key: "{{ item.1.name if item.1.gpg_key is defined else item.0.product }}"
```

Comment:
Since you created this new logic, you should cover all cases:

1. The gpg_key is defined in the product
1. The gpg_key is defined in the product and in the repository
1. The gpg_key is defined in the repository
1. The gpg_key is not defined at all